internal/server/data/identity.go

```diff
@@ -112,51 +131,329 @@ func AssignIdentityToGroups(tx GormTxn, user *models.Identity, provider *models.
 	return nil
 }
 
-func CreateIdentity(db GormTxn, identity *models.Identity) error {
-	return add(db, identity)
+func CreateIdentity(tx WriteTxn, identity *models.Identity) error {
+	if identity.VerificationToken == "" {
+		identity.VerificationToken = generate.MathRandom(10, generate.CharsetAlphaNumeric)
+	}
+	return insert(tx, (*identitiesTable)(identity))
 }
 
-func GetIdentity(db GormTxn, selectors ...SelectorFunc) (*models.Identity, error) {
-	return get[models.Identity](db, selectors...)
+type GetIdentityOptions struct {
+	ByID             uid.ID
+	ByName           string
+	PreloadGroups    bool
+	PreloadProviders bool
 }
 
-func SetIdentityVerified(db GormTxn, token string) error {
+func GetIdentity(tx GormTxn, opts GetIdentityOptions) (*models.Identity, error) {
+	if opts.ByID == 0 && opts.ByName == "" {
+		return nil, fmt.Errorf("GetIdentity must specify id or name")
+	}
+	identity := &identitiesTable{}
+	query := querybuilder.New("SELECT")
+	query.B(columnsForSelect(identity))
+	query.B("FROM")
+	query.B(identity.Table())
+	query.B("WHERE deleted_at IS NULL AND organization_id = ?", tx.OrganizationID())
+	if opts.ByID > 0 {
+		query.B("AND id = ?", opts.ByID)
+	}
+	if opts.ByName != "" {
+		query.B("AND name = ?", opts.ByName)
+	}
+
+	err := tx.QueryRow(query.String(), query.Args...).Scan(identity.ScanFields()...)
+	if err != nil {
+		return nil, handleReadError(err)
+	}
+
+	if opts.PreloadGroups {
+		groups, err := ListGroups(tx, nil, ByGroupMember(identity.ID))
+		if err != nil {
+			return nil, fmt.Errorf("load identity groups: %w", err)
+		}
+		identity.Groups = groups
+	}
+	if opts.PreloadProviders {
+		// find the providers that this identity is active in
+		opts := ListProviderUsersOptions{
+			ByIdentityID: identity.ID,
+			HideInactive: true,
+		}
+		existsInProviders, err := ListProviderUsers(tx, opts)
+		if err != nil {
+			return nil, err
+		}
+
+		var providerIDs []uid.ID
+		for _, relation := range existsInProviders {
+			providerIDs = append(providerIDs, relation.ProviderID)
+		}
+
+		providers, err := ListProviders(tx, nil, ByIDs(providerIDs))
+		if err != nil {
+			return nil, fmt.Errorf("list providers for identity: %w", err)
+		}
+		identity.Providers = providers
+	}
+
+	return (*models.Identity)(identity), nil
+}
+
+func SetIdentityVerified(tx WriteTxn, token string) error {
 	q := querybuilder.New(`UPDATE identities SET verified = true`)
-	q.B("WHERE verified = ? AND verification_token = ? AND organization_id = ?", false, token, db.OrganizationID())
+	q.B("WHERE verified = ? AND verification_token = ? AND organization_id = ?", false, token, tx.OrganizationID())
+
+	_, err := tx.Exec(q.String(), q.Args...)
+	return err
+}
+
+type ListIdentityOptions struct {
+	ByID             uid.ID
+	ByIDs            []uid.ID
+	ByNotIDs         []uid.ID
+	ByName           string
+	ByNotName        string
+	ByGroupID        uid.ID
+	CreatedBy        uid.ID
```

Comment:
In other places I think this was called `ByCreatedBy`, probably good to be consistent here, since they do the same thing.